streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableJoin.java

```diff
@@ -42,6 +44,19 @@
                 return new KTableKTableJoinValueGetter(valueGetterSupplier1.get(), valueGetterSupplier2.get());
             }
 
+            @Override
+            public String[] storeNames() {
```

Comment:
Also, you have the same block of code repeated in 4 places. Might be better to extract these into a function and only do it once.